server/src/main/java/org/elasticsearch/indices/recovery/RecoveryResponse.java

```diff
@@ -83,28 +78,16 @@ public void readFrom(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeVInt(phase1FileNames.size());
-        for (String name : phase1FileNames) {
-            out.writeString(name);
-        }
-        out.writeVInt(phase1FileSizes.size());
-        for (long size : phase1FileSizes) {
-            out.writeVLong(size);
-        }
-
-        out.writeVInt(phase1ExistingFileNames.size());
-        for (String name : phase1ExistingFileNames) {
-            out.writeString(name);
-        }
-        out.writeVInt(phase1ExistingFileSizes.size());
-        for (long size : phase1ExistingFileSizes) {
-            out.writeVLong(size);
-        }
-
+        out.writeStringList(phase1FileNames);
+        out.writeCollection(phase1FileSizes, StreamOutput::writeVLong);
+        out.writeStringList(phase1ExistingFileNames);
+        out.writeCollection(phase1ExistingFileSizes, StreamOutput::writeVLong);
         out.writeVLong(phase1TotalSize);
         out.writeVLong(phase1ExistingTotalSize);
         out.writeVLong(phase1Time);
-        out.writeVLong(phase1ThrottlingWaitTime);
+        if (out.getVersion().before(Version.V_7_0_0)) {
+            out.writeVLong(0L); // phase1ThrottlingWaitTime - not used
```

Comment:
it's not used here because it's wrongly implemented (we add the throttle time to the source shard instead of the target shard, see `createRecoverySourceHandler` in `PeerRecoverySourceService`. Instead I think it should be send back to the target shard and added there at the end of recovery.

Let's keep the `phase1ThrottlingWaitTime` here for now, and follow-up with a fix for the actual stats.

Reply:
Hmm, we send the source throttle time each FileChunk request to the target https://github.com/elastic/elasticsearch/blob/1ca6666ccab1217a07edbbc3aa89635caf3bf3b8/server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java#L610

Should `phase1ThrottlingWaitTime ` be the total throttle time on both source and target? Note that we currently use RecoveryReponse only for logging purpose.  Anyway, I added `phase1ThrottlingWaitTime` back.

Reply:
> Should phase1ThrottlingWaitTime be the total throttle time on both source and target? 

yes, we can transfer the knowledge of the source throttle time as part of the response object back to the target, and then add it to the target throttle time there. An alternative would be to transfer info about source throttle time with each file chunk that we send.